airflow/models.py

```diff
@@ -1384,7 +1404,7 @@ def _check_and_change_state_before_execution(
 
         # print status message
         self.log.info(hr + msg + hr)
-        self.try_number += 1
+        self._try_number += 1
```

Comment:
use the setter

Reply:
Using the accessor adds 2 to the value, not one. (Because the value read, which is already incremented

`self.try_number += 1` gets turned into
`self.try_number = self.try_number + 1` which gets turned into
`self._try_number = (self._try_number + 1) + 1`

Reply:
Gotcha!